### DISCUSSION TITLE
How to send proper xml ?

### DISCUSSION THREAD

**Discussion by YoannBuzenet** [vercel/next.js#54275] [Help]
### Summary

I found [doc](https://nextjs.org/docs/app/building-your-application/routing/route-handlers#non-ui-responses) about sending non-UI content with `route.ts`, but it doesn't behave like intended.

It just sends back a string and the browser doesn't see it as a XML file. It just displays text, non parsed in tags.

Should something else be added for it to be recognized as a xml file ? How would you do it ?

Thank you!
Yoann

**Comment by icyJoseph** [comment]
Have you set a content-type header, either to text/xml or application/xml?

For example:

```js
export async function GET() {
  const sitemapResponse = await fetch("https://www.sitemaps.org/sitemap.xml");

  const xml = await sitemapResponse.text();

  const response = new Response(xml, {
    status: 200,
    statusText: "ok",
  });

  response.headers.append("content-type", "text/xml");

  return response;
}
```

Do remember that there's a sitemap.ts file you can use too:

- https://nextjs.org/docs/app/api-reference/file-conventions/metadata/sitemap#generate-a-sitemap
